File: 2021/dec01/carl_toft/day1.py
import numpy as np
x = np.loadtxt("input.txt")

# Part 1
num_increasing = 0
for k in range(1, x.size):
    if x[k] > x[k-1]:
        num_increasing = num_increasing + 1
print("Part 1:", num_increasing)

# Part 2
num_increasing = 0
for k in range(1, x.size-2):
    if np.sum(x[k:k+3]) > np.sum(x[k-1:k+2]):
        num_increasing = num_increasing + 1
print("Part 2:", num_increasing)
xxx = 3

File: 2022/dec01/joakim_loefgren/day1.lua
local sum = 0
-- holds calorie counts for the top elves
-- use top = {0} for part I
local top = {0, 0, 0}

for line in io.lines('input.txt') do
    local cal = tonumber(line)
    if cal then
        sum = sum + cal
    else
        for i, val in ipairs(top) do
            if sum > val then
                for j = #top - i + 1, i + 1, -1 do
                    top[j] = top[j - 1]
                end
                top[i] = sum
                break
            end
        end
        sum = 0
    end
end

-- handle the last elf
for i, val in ipairs(top) do
    if sum > val then
        for j = #top - i + 1, i + 1, -1 do
            top[j] = top[j - 1]
        end
        top[i] = sum
        break
    end
end

local tot = 0
for _, cal in ipairs(top) do
    tot = tot + cal
end
print(tot)
